gradle/scripts/lib/java-javadoc.gradle

```diff
@@ -0,0 +1,154 @@
+              trimLines(lines);
+              for (var j in lines) {
+                lines[j] = lines[j].replace(/^> ?/, "");
+              }
+              trimLines(lines);
+              child.textContent = lines.join("\\n");
+
+              if (hasHljs) {
+                hljs.highlightBlock(child);
+              }
+            }
+            </script>'''.readLines().stream()
+                    .map({ line -> line.trim() })
+                    .filter({ line -> !line.isEmpty() && !line.startsWith('//') })
+                    .collect().join('')
+
+            encoding = 'UTF-8'
+            docEncoding = 'UTF-8'
+            breakIterator = true
+            version = false
+            author = false
+            keyWords = false
+            quiet()
+            stylesheetFile = defaultStylesheetFile
+
+            // Enable all lints except the missing tag warnings
+            addBooleanOption('Xdoclint:all').value = true
+            addBooleanOption('Xdoclint:-missing').value = true
+
+            // Add --allow-script-in-comments if available (since 1.8.0_121)
+            try {
+                if (Class.forName('com.sun.tools.doclets.formats.html.ConfigurationImpl')
+                        .newInstance().optionLength('--allow-script-in-comments') > 0) {
+                    addBooleanOption('-allow-script-in-comments').value = true
+                }
+            } catch (ignored) {}
+
+            // External Javadoc links, cached under '~/.gradle/caches/package-lists'
+            def addOfflineLink = { name, url ->
+                def javadocUrl = "${url}"
+                if (!javadocUrl.endsWith('/')) {
+                    javadocUrl += '/'
+                }
+
+                def javadocUrlSha1 = MessageDigest.getInstance('SHA1').digest(javadocUrl.getBytes('UTF-8')).encodeHex()
+                def packageListFile = new File(javadocCacheDir, "${name}/${javadocUrlSha1}/package-list")
+                if (!packageListFile.exists()) {
+                    packageListFile.parentFile.mkdirs()
+                    def packageListUrl = new URL("${javadocUrl}package-list")
+                    logger.lifecycle("Downloading: ${packageListUrl}")
+                    try {
+                        // Set some fake headers for the web sites who blocks a URLConnection.
+                        def conn = packageListUrl.openConnection() as HttpURLConnection
+                        conn.setRequestProperty("Accept", "text/html,application/xhtml+xml,application/xml;q=0.9,*/*;q=0.8")
+                        conn.setRequestProperty('Accept-Encoding', 'identity')
+                        conn.setRequestProperty("Accept-Language", 'en-US,en;q=0.5')
+                        conn.setRequestProperty("Cache-Control", 'no-cache')
+                        conn.setRequestProperty('Pragma', 'no-cache')
+                        conn.setRequestProperty('User-Agent', 'Mozilla/5.0 (X11; Linux x86_64; rv:57.0) Gecko/20100101 Firefox/57.0')
```

Comment:
Should probably mention armeria here

Reply:
Changed the user agent to `Gradle/<gradle_version> (<groupId>:<artifactId>)`